stdlib/ephemeron.mli

```diff
@@ -75,43 +75,106 @@ module type S = sig
+    [@@alert old_ephemeron_api "This function won't be available in 5.0"]
+  val to_seq_values : 'a t -> 'a Seq.t
+    [@@alert old_ephemeron_api "This function won't be available in 5.0"]
+  val add_seq : 'a t -> (key * 'a) Seq.t -> unit
+  val replace_seq : 'a t -> (key * 'a) Seq.t -> unit
+  val of_seq : (key * 'a) Seq.t -> 'a t
 
   val clean: 'a t -> unit
   (** remove all dead bindings. Done automatically during automatic resizing. *)
 
   val stats_alive: 'a t -> Hashtbl.statistics
   (** same as {!Hashtbl.SeededS.stats} but only count the alive bindings *)
 end
-(** The output signature of the functor {!K1.Make} and {!K2.Make}.
+(** The output signature of the functors {!K1.Make} and {!K2.Make}.
     These hash tables are weak in the keys. If all the keys of a binding are
     alive the binding is kept, but if one of the keys of the binding
     is dead then the binding is removed.
 *)
 
 module type SeededS = sig
-  include Hashtbl.SeededS
+
+  type key
+  type !'a t
+  val create : ?random (*thwart tools/sync_stdlib_docs*) : bool -> int -> 'a t
+  val clear : 'a t -> unit
+  val reset : 'a t -> unit
+  val copy : 'a t -> 'a t
+  val add : 'a t -> key -> 'a -> unit
+  val remove : 'a t -> key -> unit
+  val find : 'a t -> key -> 'a
+  val find_opt : 'a t -> key -> 'a option
+  val find_all : 'a t -> key -> 'a list
+  val replace : 'a t -> key -> 'a -> unit
+  val mem : 'a t -> key -> bool
+  val iter : (key -> 'a -> unit) -> 'a t -> unit
+    [@@alert old_ephemeron_api "This function won't be available in 5.0"]
+  val filter_map_inplace : (key -> 'a -> 'a option) -> 'a t -> unit
+    [@@alert old_ephemeron_api "This function won't be available in 5.0"]
+  val fold : (key -> 'a -> 'b -> 'b) -> 'a t -> 'b -> 'b
+    [@@alert old_ephemeron_api "This function won't be available in 5.0"]
+  val length : 'a t -> int
+  val stats : 'a t -> Hashtbl.statistics
+  val to_seq : 'a t -> (key * 'a) Seq.t
+    [@@alert old_ephemeron_api "This function won't be available in 5.0"]
+  val to_seq_keys : _ t -> key Seq.t
+    [@@alert old_ephemeron_api "This function won't be available in 5.0"]
+  val to_seq_values : 'a t -> 'a Seq.t
+    [@@alert old_ephemeron_api "This function won't be available in 5.0"]
+  val add_seq : 'a t -> (key * 'a) Seq.t -> unit
+  val replace_seq : 'a t -> (key * 'a) Seq.t -> unit
+  val of_seq : (key * 'a) Seq.t -> 'a t
+
   val clean: 'a t -> unit
   (** remove all dead bindings. Done automatically during automatic resizing. *)
 
   val stats_alive: 'a t -> Hashtbl.statistics
   (** same as {!Hashtbl.SeededS.stats} but only count the alive bindings *)
 end
-(** The output signature of the functor {!K1.MakeSeeded} and {!K2.MakeSeeded}.
+(** The output signature of the functors {!K1.MakeSeeded} and {!K2.MakeSeeded}.
 *)
 
 module K1 : sig
   type ('k,'d) t (** an ephemeron with one key *)
 
   val create: unit -> ('k,'d) t
+    [@@alert old_ephemeron_api "This function won't be available in 5.0"]
```

Comment:
Why use a new alert instead of using the already existing `deprecated` alert? It seems rather fitting to use it here.

Reply:
I'm on the fence. On the one hand you're right; on the other hand we had bad expericences with premature deprecation breaking packages (I'm thinking of `Pervasives`). Since there are not that many users of ephemerons, I could be convince to switch to `deprecated`. Does anyone else have an opinion?

Reply:
Probably doesn't make much of a difference given the relatively few Ephemeron users, but I think using an ad-hoc alert is slightly better as it won't be made fatal by Dune. Here we are introducing the replacement of an API at the same time as we are deprecating it which is a bit unfriendly in general...